src/libraries/Microsoft.Extensions.Caching.Abstractions/ref/Microsoft.Extensions.Caching.Abstractions.csproj

```diff
@@ -6,6 +6,10 @@
 
   <ItemGroup>
     <Compile Include="Microsoft.Extensions.Caching.Abstractions.cs" />
+    <Compile Include="Microsoft.Extensions.Caching.Abstractions.net7.0.cs" Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
```

Comment:
Why not also allow the DIM on net6?

Reply:
It seems to be what we approved it with arbitrarily with `#if NET_7_OR_GREATER` over the API review meeting.

Reply:
I'm not sure how much it matters, and maybe we can wait for feedback, but I guess I would have added it for all the .NETCOREAPP TFMs we support. That way if someone was on the LTS .NET 6, but using the 7.0 MemoryCache nuget package, they could use this API.